src/Adapter/Image/ImageRetriever.php

```diff
@@ -54,10 +59,12 @@ public function getProductImages(array $product, Language $language)
                 $image['associatedVariants'] = [];
             }
 
-            return $image;
+            if (in_array($productAttributeId, $image['associatedVariants'])) {
```

Comment:
the interesting point is here: we shouldnt return all images when we have a product attribute id :)